pubsublite/pscompat/publisher.go

```diff
@@ -99,7 +106,15 @@ func NewPublisherClientWithSettings(ctx context.Context, topic string, settings
 	if err := wirePub.WaitStarted(); err != nil {
 		return nil, err
 	}
-	return &PublisherClient{settings: settings, wirePub: wirePub}, nil
+	publisher := &PublisherClient{settings: settings, wirePub: wirePub}
+
+	// Mitigation for Stop not being called when the publisher client is no longer
+	// used. Users must still call Stop to promptly shut down the publisher, as
+	// finalizers run after an arbitrary amount of time.
+	runtime.SetFinalizer(publisher, func(p *PublisherClient) {
+		p.wirePub.Stop()
```

Comment:
I would do "go p.wirePub.Stop()" to get it out of the garbage collector thread

Reply:
Done